clang/include/clang/Driver/Options.td

```diff
@@ -7297,7 +7297,7 @@ defm sycl_allow_device_image_dependencies: BoolOptionWithoutMarshalling<"f", "sy
   PosFlag<SetTrue, [], [ClangOption], "Allow dependencies between device code images">,
   NegFlag<SetFalse, [], [ClangOption], "Do not allow dependencies between device code images (default)">>;
 
-def fsycl_dump_device_code_EQ : Joined<["-"], "fsycl-dump-device-code=">,
+def fdump_device_code_EQ : Joined<["-"], "fdump-device-code=">,
```

Comment:
To reduce the impact to users that are already using `-fsycl-dump-device-code` we should first deprecate this option and add `-fdump-device-code`.  `-fsycl-dump-device-code` would then be an alias.  This deprecation should also emit a diagnostic on what the preferred option name to use is.  Usage would be something like what `-fsycl-use-bitcode` does.